test/selenium_tests/test_custom_builds.py

```diff
@@ -0,0 +1,107 @@
+        super(CustomBuildsTestcase, self).setUp()
+        self.ensure_user()
+
+    @selenium_test
+    def test_build_add(self):
+        self.navigate_to_custom_builds_page()
+        self.add_custom_build(self.build_name1, self.build_key1)
+        self.assert_custom_builds_in_grid([self.build_name1])
+
+    @selenium_test
+    def test_build_delete(self):
+        self.navigate_to_custom_builds_page()
+
+        self.add_custom_build(self.build_name2, self.build_key2)
+        self.assert_custom_builds_in_grid([self.build_name2])
+
+        self.delete_custom_build(self.build_name2)
+        self.assert_custom_builds_in_grid([self.build_name2], False)
+
+    @retry_assertion_during_transitions
+    def assert_custom_builds_in_grid(self, expected_builds, present=True):
+        actual_builds = self.get_custom_builds()
+        intersection = set(actual_builds).intersection(expected_builds)
+        if present:
+            self.assertEqual(intersection, set(expected_builds))
+        else:
+            self.assertEqual(intersection, set())
+
+    def add_custom_build(self, build_name, build_key):
+        name_div = self.wait_for_selector('div[tour_id="name"')
+        name_input = name_div.find_element_by_css_selector('.ui-form-field input')
+        name_input.click()
+        name_input.send_keys(build_name)
+
+        key_div = self.wait_for_selector('div[tour_id="id"')
+        key_input = key_div.find_element_by_css_selector('.ui-form-field input')
+        key_input.click()
+        key_input.send_keys(build_key)
+
+        len_type_div = self.wait_for_selector('div[tour_id="len|type')
+        len_type_select = len_type_div.find_element_by_css_selector('.ui-form-field .ui-select')
+        len_type_select.click()
+
+        option = self.wait_for_sizzle_selector_clickable('div[role="option"]:contains("Len-file by copy/paste")')
+        option.click()
+        content_area = self.wait_for_and_click_selector('.ui-form-field > textarea')
+        content_area.send_keys('content')
+
+        self.wait_for_and_click_selector('button#save')
+
+    def delete_custom_build(self, build_name):
+        delete_button = None
+        grid = self.wait_for_selector('table.grid > tbody')
+        for row in grid.find_elements_by_tag_name('tr'):
+            td = row.find_elements_by_tag_name('td')
+            name = td[0].text
+            if name == build_name:
+                delete_button = td[3].find_element_by_css_selector('.ui-button-icon-plain')
+                break
+
+        if delete_button is None:
+            raise AssertionError('Failed to find custom build with name [%s]' % build_name)
+
+        delete_button.click()
+
+    def get_custom_builds(self):
+        time.sleep(1)
+        builds = []
+        grid = self.wait_for_selector('table.grid > tbody')
+        for row in grid.find_elements_by_tag_name('tr'):
+            td = row.find_elements_by_tag_name('td')
+            name = td[0].text
+            builds.append(name)
```

Comment:
Can we merge the lines from 83-85 into one?

`builds.append(row.find_elements_by_tag_name('td')[0].text)`

`td` and `name` are not being used in the method anywhere else.